packages/ui/src/forms/fieldSchemasToFormState/renderField.tsx

```diff
@@ -126,17 +129,18 @@ export const renderField: RenderFieldMethod = ({
 
           row.customComponents.RowLabel = !mockRSCs
             ? RenderServerComponent({
-                clientProps,
-                Component: fieldConfig.admin.components.RowLabel,
-                importMap: req.payload.importMap,
-                key: `${rowIndex}`,
-                serverProps: {
-                  ...serverProps,
+                clientProps: {
+                  ...(clientProps as Partial<ArrayRowLabelClientProps> satisfies Partial<ArrayRowLabelClientProps>),
                   rowLabel: `${getTranslation(fieldConfig.labels.singular, req.i18n)} ${String(
                     rowIndex + 1,
                   ).padStart(2, '0')}`,
                   rowNumber: rowIndex + 1,
-                },
+                } satisfies Partial<ArrayRowLabelClientProps>,
+                Component: fieldConfig.admin.components.RowLabel,
+                importMap: req.payload.importMap,
+                key: `${rowIndex}`,
+                serverProps:
+                  serverProps as Partial<ArrayRowLabelServerProps> satisfies Partial<ArrayRowLabelServerProps>,
```

Comment:
This is the use case for something like `ArrayRowLabelServerPropsOnly`:

```suggestion
                  serverProps satisfies ArrayRowLabelServerProps
```

Reply:
I made specific types for server props only for both array and block row labels.